go/libraries/doltcore/sqle/dolt_index_test.go

```diff
@@ -0,0 +1,637 @@
+		},
+		{
+			"onepk:idx_v1",
+			[]interface{}{2},
+			[]interface{}{4},
+			[]sql.Row{{3, 3, 3}, {4, 4, 3}},
+		},
+		{
+			"onepk:idx_v1",
+			[]interface{}{1},
+			[]interface{}{4},
+			[]sql.Row{{1, 1, 1}, {2, 1, 2}, {3, 3, 3}, {4, 4, 3}},
+		},
+		{
+			"twopk:primaryKey",
+			[]interface{}{1, 1},
+			[]interface{}{1, 1},
+			[]sql.Row{{1, 1, 3, 3}},
+		},
+		{
+			"twopk:primaryKey",
+			[]interface{}{1, 1},
+			[]interface{}{2, 1},
+			[]sql.Row{{1, 1, 3, 3}, {1, 2, 3, 4}, {2, 1, 4, 4}},
+		},
+		{
+			"twopk:primaryKey",
+			[]interface{}{1, 1},
+			[]interface{}{2, 5},
+			[]sql.Row{{1, 1, 3, 3}, {1, 2, 3, 4}, {2, 1, 4, 4}, {2, 2, 4, 3}},
+		},
+		{
+			"twopk:idx_v2v1",
+			[]interface{}{3, 3},
+			[]interface{}{3, 4},
+			[]sql.Row{{1, 1, 3, 3}, {2, 2, 4, 3}},
+		},
+		{
+			"twopk:idx_v2v1",
+			[]interface{}{3, 4},
+			[]interface{}{4, 4},
+			[]sql.Row{{1, 2, 3, 4}, {2, 1, 4, 4}, {2, 2, 4, 3}},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("%s|%v%v", test.indexName, test.greaterThanOrEqual, test.lessThanOrEqual), func(t *testing.T) {
+			index, ok := indexMap[test.indexName]
+			require.True(t, ok)
+
+			expectedRows := convertSqlRowToInt64(test.expectedRows)
+
+			indexLookup, err := index.AscendRange(test.greaterThanOrEqual, test.lessThanOrEqual)
+			require.NoError(t, err)
+			dil, ok := indexLookup.(*doltIndexLookup)
+			require.True(t, ok)
+			indexIter, err := dil.RowIter(NewTestSQLCtx(context.Background()))
+			require.NoError(t, err)
+
+			var readRows []sql.Row
+			var nextRow sql.Row
+			for nextRow, err = indexIter.Next(); err == nil; nextRow, err = indexIter.Next() {
+				readRows = append(readRows, nextRow)
+			}
+			require.Equal(t, io.EOF, err)
+
+			assert.ElementsMatch(t, expectedRows, readRows)
+
+			indexLookup, err = index.DescendRange(test.lessThanOrEqual, test.greaterThanOrEqual)
+			require.NoError(t, err)
+			dil, ok = indexLookup.(*doltIndexLookup)
+			require.True(t, ok)
+			indexIter, err = dil.RowIter(NewTestSQLCtx(context.Background()))
+			require.NoError(t, err)
+
+			readRows = nil
+			for nextRow, err = indexIter.Next(); err == nil; nextRow, err = indexIter.Next() {
+				readRows = append(readRows, nextRow)
+			}
+			require.Equal(t, io.EOF, err)
+
+			assert.ElementsMatch(t, expectedRows, readRows)
+		})
+	}
+}
+
+func testDoltIndex(t *testing.T, keys []interface{}, expectedRows []sql.Row, compFunc func(keys ...interface{}) (sql.IndexLookup, error)) {
+	indexLookup, err := compFunc(keys...)
+	require.NoError(t, err)
+	dil, ok := indexLookup.(*doltIndexLookup)
+	require.True(t, ok)
+	indexIter, err := dil.RowIter(NewTestSQLCtx(context.Background()))
+	require.NoError(t, err)
+
+	var readRows []sql.Row
+	var nextRow sql.Row
+	for nextRow, err = indexIter.Next(); err == nil; nextRow, err = indexIter.Next() {
+		readRows = append(readRows, nextRow)
+	}
+	require.Equal(t, io.EOF, err)
+
+	assert.ElementsMatch(t, convertSqlRowToInt64(expectedRows), readRows)
```

Comment:
Order is deterministic at this point. I went this way because of the earlier conversations regarding the `ORDER BY` stuff, but I guess it's more useful to assert the expected order here too since that's further up the chain.